File: wccapp/src/app/service/customer.service.ts
import {Injectable} from '@angular/core';
import {environment} from '../../environment/environment';
import {HttpClient, HttpHeaders} from '@angular/common/http';
import {Observable} from 'rxjs';
import {Customer} from '../model/account-info';

@Injectable({
  providedIn: 'root'
})
export class CustomerService {

  private apiServerUrl = environment.apiBaseUrl + '/wcc/customer';


  constructor(private http: HttpClient) {
  }

  public login(customer: Customer): Observable<Customer> {
    return this.http.post<Customer>(`${this.apiServerUrl}/login`, customer);
  }

  public logout(): Observable<string> {
    return this.http.post<string>(`${this.apiServerUrl}/logout`, {});
  }

  public addCustomer(customer: Customer): Observable<Customer> {
    return this.http.post<Customer>(`${this.apiServerUrl}/add`, customer);
  }

  public getCustomer(userId: string): Observable<Customer> {
    return this.http.get<Customer>(`${this.apiServerUrl}/findByUserId`, {
      params: { userId }
    });
  }

  public updateCustomer(customer: Customer): Observable<Customer> {
    return this.http.put<Customer>(`${this.apiServerUrl}/update`, customer);
  }

}

File: wcc/src/main/java/com/intern/wcc/service/CustomerService.java
package com.intern.wcc.service;

import com.intern.wcc.entity.Customer;
import com.intern.wcc.model.helper.SearchModel;
import jakarta.servlet.http.HttpServletRequest;

import java.util.List;

public interface CustomerService {

    Customer login(HttpServletRequest request, Customer customer);

    String logout(HttpServletRequest request);

    List<Customer> findAllCustomer(SearchModel searchModel);

    Customer findByUserId(String userId);

    boolean addCustomer(Customer customer);

    Customer updateCustomer(Customer customer);
}
